### ISSUE TITLE
add Direct Boot support for Android so RHVoice can be used with TalkBack before the initial unlock

### ISSUE THREAD

**Issue by thestinger** [RHVoice/RHVoice#271]
More information:

https://developer.android.com/training/articles/direct-boot

https://github.com/GrapheneOS/platform_packages_apps_Updater is a trivial example of using this.

You need to mark a subset of the functionality as direct boot aware and make sure it's capable of running in that state. A minimal amount of data needs to be migrated to being stored in device encrypted storage via the APIs for this. No sensitive user data should be marked that way. The voice setup and other configuration is what's expected to be there. There shouldn't be any kind of history, stats or other sensitive data in device encrypted storage though.

The open source release of TalkBack is available here and has Direct Boot support:

https://github.com/google/talkback

This is wanted for GrapheneOS and we'd like to integrate RHVoice as a standard text-to-speech implementation. However, we currently lack the resources to implement this ourselves and to refine other aspects of it to make it suitable for inclusion. We're going to be shipping the open source TalkBack though and it'd be really nice to have a TTS implementation to go along with it. We'd also need to more carefully review the security of RHVoice since unlike TalkBack it would be enabled by default.

**Comment by PatrykMis** [comment]
@alex19EP I think it should be rated high, not medium and as very important enhancement. Currently, there is no open source TTS which has this feature.
For example, as a blind human beeing, currently I'm using Lineage with MicroG+Talkback built from source and I can't (and don't want to) use Google Text-To-Speech mainly because is closed proprietary software. So my biggest issue is when I boot my OS and require sighted assistance to enter my pin and then TTS starts.

**Comment by PatrykMis** [comment]
https://www.bountysource.com/issues/97074123-add-direct-boot-support-for-android-so-rhvoice-can-be-used-with-talkback-before-the-initial-unlock

**Comment by PatrykMis** [comment]
Anyone to fix this issue? I am ready for extensive testing. It would be first open-sourced TTS that supports this functionality on Android. Looking forward to see this fixed here as Android source looks well-maintained - a lot better than e.g. ESpeakNG.

**Comment by shuteye4220** [comment]
Hello: bumping this issue up again... I too am a blind user, ironically of Graphene OS and would love to be able to use RHVoice at first boot: otherwise, I'm not able to enter in my own pin number after rebooting my phone without sighted help. Please add first boot permitions to RHVoice as above :) Happy to help in any way, though I'm not a coder unfortunately.

**Comment by PatrykMis** [comment]
@shuteye4220 use current development version of eSpeak, it's a workarround. It'll work on boot then TalkBack switches to chosen TTS (RhVoice in this case) after receiving `BOOT_UNLOCKED`. If you can't compile yourself drop me an e-mail at foss@patrykmis.com; I'd rather prefer not to put links to binaries here in the comment, though compiling eSpeak for Android is pretty straight forward.

**Comment by shuteye4220** [comment]
Thanks for the advice Patryk - sent you an email.

**Comment by shuteye4220** [comment]
Hi Patryk, haven't yet received an answer - is this the guide I should use to compile ESpeak as per your above suggestion? (I'm so not a coder but really want to get my phone working without needing sighted assistance every time I reboot) https://github.com/espeak-ng/espeak-ng/blob/master/docs/building.md
